src/containers/share-modal/components/ShareDrawer.tsx

```diff
@@ -1,117 +1,66 @@
-import { ToastContext } from 'components/toast/ToastContext'
-import { make, useRecord } from 'store/analytics/actions'
-import { getCopyableLink } from 'utils/clipboardUtil'
-import { SHARE_TOAST_TIMEOUT_MILLIS } from 'utils/constants'
 import { isDarkMode } from 'utils/theme/theme'
-import { openTwitterLink } from 'utils/tweet'
 
 import { messages } from '../messages'
+import { ShareProps } from '../types'
 
+import { IconTikTok } from './IconTikTok'
 import styles from './ShareDrawer.module.css'
 
-export const ShareDrawer = () => {
-  const [isOpen, setIsOpen] = useModalState('Share')
-  const { toast } = useContext(ToastContext)
-  const dispatch = useDispatch()
-  const track = useSelector(getTrack)
-  const artist = useSelector((state: CommonState) =>
-    getUser(state, { id: track?.owner_id })
-  )
-  const account = useSelector(getAccountUser)
-  const source = useSelector(getSource)
-  const record = useRecord()
+type ShareDrawerProps = ShareProps
 
+export const ShareDrawer = ({
+  onShareToTwitter,
+  onShareToTikTok,
+  onCopyLink,
+  isOpen,
+  onClose,
+  isOwner
+}: ShareDrawerProps) => {
   const getActions = useCallback(() => {
-    const isOwner = account && artist && account.user_id === artist.user_id
-
     const shareToTwitterAction = {
-      text: messages.twitter,
       icon: <IconTwitterBird height={20} width={26} />,
+      text: messages.twitter,
       className: styles.shareToTwitterAction,
-      onClick: () => {
-        if (track && artist && source) {
-          const twitterText = `Check out ${track.title} by ${artist.handle} on @AudiusProject #Audius`
-          const trackLink = getCopyableLink(track.permalink)
-          openTwitterLink(trackLink, twitterText)
-          record(
-            make(Name.SHARE_TO_TWITTER, {
-              kind: 'track',
-              source,
-              id: track.track_id,
-              url: trackLink
-            })
-          )
-        } else {
-          console.error(
-            'Tried to share a track to twitter, but track and/or artist was missing'
-          )
-        }
-      }
+      onClick: onShareToTwitter
     }
 
     const shareToTikTokAction = {
       text: messages.tikTok,
-      icon: isDarkMode() ? (
-        <IconTikTokInverted height={26} width={26} />
-      ) : (
-        <IconTikTok height={26} width={26} />
-      ),
-      onClick: () => {
-        if (track) {
-          setIsOpen(false)
-          dispatch(requestOpenTikTokModal({ id: track.track_id }))
-        } else {
-          console.error('Tried to share sound to TikTok but track was missing')
-        }
-      }
+      icon: <IconTikTok height={32} width={32} />,
+      className: cs(styles.shareToTikTokAction, {
+        [styles.shareToTikTokActionDark]: isDarkMode()
+      }),
+      onClick: onShareToTikTok
     }
 
     const copyLinkAction = {
-      text: messages.copyLink,
-      icon: <IconLink height={26} width={26} />,
+      text: 'Copy Link to Track',
```

Comment:
Can we pull this string up to a `messages` object for easier localization in the future?